libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -336,6 +337,68 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return crypto.createHash('sha1').update(shrinkwrapContent).digest('hex');
   }
 
+  /**
+   * Determine whether `pnpm-lock.yaml` contains insecure sha1 hashes.
+   * @internal
+   */
+  private _disallowInsecureSha1(
+    customTipsConfiguration: CustomTipsConfiguration,
+    terminal: ITerminal
+  ): boolean {
+    const { packages } = this._shrinkwrapJson;
+    if (packages) {
+      for (const { resolution } of Object.values(packages)) {
+        if ((resolution as { integrity: string }).integrity.startsWith('sha1')) {
```

Comment:
If the resolution property is missing, this will fail.

```suggestion
        if ((resolution?.integrity.startsWith('sha1')) {
```

This type is

```
  /** Information about the resolved package */
  resolution?: {
    /** The hash of the tarball, to ensure archive integrity */
    integrity: string;
    /** The name of the tarball, if this was from a TGX file */
    tarball?: string;
  };
```